bundles/org.openhab.core.io.rest/src/main/java/org/eclipse/smarthome/io/rest/internal/resources/RootResource.java

```diff
@@ -54,17 +68,29 @@
  * @author Kai Kreuzer - Initial contribution and API
  */
 @Path("/")
-@Component(service = RootResource.class, configurationPid = "org.eclipse.smarthome.io.rest.root")
-public class RootResource {
+@NonNullByDefault
+@Component(service = { RootResource.class, RESTservice.class }, configurationPid = "org.eclipse.smarthome.io.rest.root")
+public class RootResource implements RESTservice {
 
     private final transient Logger logger = LoggerFactory.getLogger(RootResource.class);
 
     private List<RESTResource> restResources = new ArrayList<RESTResource>();
 
-    private ConfigurationAdmin configurationAdmin;
+    protected @NonNullByDefault({}) @Context UriInfo uriInfo;
+    private @NonNullByDefault({}) String restRoot;
+
+    @Reference
+    protected @NonNullByDefault({}) HttpService httpService;
+    private @NonNullByDefault({}) ServletContainer servletContainer;
+
+    private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
```

Comment:
This adds a permanent additional (and during runtime idle) thread. I'd prefer to instead use the [common thread pool](https://github.com/openhab/openhab-core/blob/48d873a32ba0901a9b58f04c541f0f858fbfcd51/bundles/org.openhab.core/src/main/java/org/eclipse/smarthome/core/common/ThreadPoolManager.java#L55-L60) for this job.